src/main.zig

```diff
@@ -130,8 +132,14 @@ const BranchNode = struct {
     depth: u8,
     count: u8,
     crs: *CRS,
+    commitment: ?*Element,
 
     fn computeCommitment(self: *const BranchNode) anyerror!Element {
+        if (self.commitment != null) {
+            // TODO return a pointer instead
```

Comment:
this is a note to self, that I don't want to address in this PR but in a subsequent one, as this will require an extensive rewrite and this PR is complex enough as it is.